### PR TITLE
Sm/perf scale nuts

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
record some perf tests for baselining how SDR handles larger repos across OSes

### What issues does this PR fix or reference?
it's prework for [@W-11492994@](https://gus.lightning.force.com/a07EE000012HtqzYAC)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "node": ">=14.0.0"
   },
   "dependencies": {
+    "@salesforce/cli-plugins-testkit": "^2.3.4",
```
Comment (WillieRuemmele) [inline package.json:28]: should this be a `dev-dep`?

File: .vscode/settings.json [hunk 2]
```diff
@@ -15,5 +15,6 @@
   "eslint.format.enable": true,
   "editor.codeActionsOnSave": {
     "source.fixAll": true
-  }
+  },
```
Comment (WillieRuemmele) [inline .vscode/settings.json:18]: is this required?

File: package.json (hunk 1 above)
Comment (mshanemc) [reply package.json:28]: yes

File: .vscode/settings.json (hunk 2 above)
Comment (mshanemc) [reply .vscode/settings.json:18]: I think the linter just likes trailing commas 🤷🏻

File: test/nuts/scale/lotsOfClasses.nut.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      const dirName = path.join(classdir, `dir${d}`);
+      // eslint-disable-next-line no-await-in-loop
+      await fs.promises.mkdir(dirName);
+      for (let c = 0; c < classesPerDir; c++) {
+        const className = `x${d}x${c}`;
+        // eslint-disable-next-line no-await-in-loop
+        await Promise.all([
+          fs.promises.writeFile(
+            path.join(dirName, `${className}.cls`),
+            `public with sharing class ${className} {public ${className}() {}}`
```
Comment (shetzel) [inline test/nuts/scale/lotsOfClasses.nut.ts:47]: Would it be interesting to have a class that is more typical in size?  Maybe somewhere between 100-200 lines?

File: test/nuts/scale/lotsOfClasses.nut.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+    const components = resolver.getComponentsFromPath(path.join(session.project.dir, 'mdapiOut'));
+    await converter.convert(components, 'source', {
+      type: 'directory',
+      outputDirectory: path.join(session.project.dir, 'mdapiOut'),
+      packageName: 'SourceFormatPackage',
+    });
+    performance.measure('mdapiToSource', 'MdapiToSource');
+  });
+  it('convert zip to source', async () => {});
```
Comment (shetzel) [inline test/nuts/scale/lotsOfClasses.nut.ts:114]: Did you mean to implement this one?

**Review by shetzel — APPROVED** [review]
Looks good.  Is it worth measuring the manifest creation code?  Not i/o intensive in any way, and maybe not a significant source of cpu time, but it's potentially interesting.

File: test/nuts/scale/lotsOfClasses.nut.ts (hunk 4 above)
Comment (mshanemc) [reply test/nuts/scale/lotsOfClasses.nut.ts:114]: I left it for feedback.  I'm not sure what's interesting to do now vs. later vs. not (vs. getting some baselines and moving on to the actual problem this is all pre-work for)

Comment (mshanemc) [reply test/nuts/scale/lotsOfClasses.nut.ts:114]: digging around sdr, it looks like unzip streams to convert anyway (like an alternate reader).  I'll remove for now.

**Comment by mshanemc** [comment]
all these changes are in the https://github.com/forcedotcom/source-deploy-retrieve/pull/683.  merging this would block stuff without the windows fix.

**Pull request closed without merging**